### ISSUE TITLE
integrate pcap capture into debug configure flag

### ISSUE THREAD

**Issue by flihp** [tpm2-software/tpm2-tss#860]
Being able to decode command / response buffers requires that we're able to capture them in the first place. My first instinct is to put this into the TCTI modules but this means it will have to be duplicated across all TCTIs. So maybe it's better to put this into the `tss2-sys` library?

In either case it shouldn't be enabled by default and probably should only be enabled when the configure script is provided with the `--debug` flag or something equivalent.

**Comment by tstruk** [comment]
Alternatively we can capture the packet with a generic tool like tcpdump. This should do the job:
sudo tcpdump -i lo 'tcp port 2321 or 2322' -w dump.pcap

**Comment by martinezjavier** [comment]
@tstruk that would work for the simulator, but not for example to capture command / response buffers when using the device TCTI.

@flihp which makes me wonder, does pcap even support capturing from chardevs or only from network interfaces? If the answer is no, then I think it belongs to the TCTI layer, since how to capture data will depend on the transmission interface used?

**Comment by AndreasFuchsSIT** [comment]
I'd recommend creating a new TCTI module that behaves similarly to the proxy tcti from https://github.com/tpm2-software/tpm2-tss/blob/master/test/integration/main-esapi.c#L22

I guess the proxy_tcti_conf could be "<innter_tcti_name>:<innter_tcti_conf>" for this more generic module.

**Comment by flihp** [comment]
I don't know a whole lot about libpcap which may be obvious here. I'm just throwing out ideas and trying to organize what I think the tasks will be in the issue tracker. Seems like the right place to collect data even if it's wrong at first.

I'm going to play around with libpcap over in https://github.com/tpm2-software/tpm2-abrmd/milestone/8 to see what I can come up with. Having the decoder from the work @tstruk is doing in wireshark will be necessary for testing though.

**Comment by joholl** [comment]
Oops. I didn't notice that this issue exists, but I implemented tcti-pcap already: #1817

It is basically a TCTI module which logs the TPM traffic in pcap-ng format and forwards it to another TCTI module.

```
          +-----------+     +-----------+
SAPI ==>  | tcti-pcap | ==> | tcti-...  | ==> ...
          +-----------+     +-----------+
                ||
                \/
          tpm2_log.pcap
```

**Comment by AndreasFuchsSIT** [comment]
tcti-pcap was add in #1817
